app/src/main/java/com/arctouch/codechallenge/home/HomeAdapter.kt

```diff
@@ -34,7 +85,12 @@ class HomeAdapter(private val movies: List<Movie>) : RecyclerView.Adapter<HomeAd
         return ViewHolder(view)
     }
 
-    override fun getItemCount() = movies.size
+    override fun getItemCount() = if (movies == null) 0 else movies!!.size
```

Comment:
Here you should use the Elvis operator `?:` to execute this same verification:

    override fun getItemCount() = movies?.let{ movies.size } ?: 0

The left side of `?:` will be evaluated if `movies` are not null. Otherwise, it will be evaluated as 0.

Reply:
Correct, thank you again!